### PR TITLE
docs: Update documentation to reflect SSO unification in backward compatible way

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
 /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
Updates operator documentation to inform users about Changes to `.spec.sso` and new SSO configuration method. Also informs users of deprecation notice for `.spec.dex`, `.spec.sso` keycloak fields and `DISABLE_DEX`
Updates API spec
Re formats `api.html.md` to make it more readable while contributing

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/reference/argocd.md [hunk 1]
```diff
@@ -1167,34 +1224,21 @@ spec:
...
+    `.spec.sso.Image`, `.spec.sso.Version`, `.spec.sso.Resources` and `.spec.sso.verifyTLS` are deprecated and support will be removed in Argo CD operator v0.6.0. Please use equivalent fields under `.spec.sso.keycloak` to configure your keycloak instance.
+
 The following properties are available for configuring the Single sign-on component.
 
 Name | Default | Description
 --- | --- | ---
 Image | OpenShift - `registry.redhat.io/rh-sso-7/sso75-openshift-rhel8` <br/> Kuberentes - `quay.io/keycloak/keycloak` | The container image for keycloak. This overrides the `ARGOCD_KEYCLOAK_IMAGE` environment variable.
-Provider | [Empty] | The name of the provider used to configure Single sign-on. For now the only supported option is keycloak.
+[Keycloak](#keycloak-options) | [Object] | Configuration options for Keycloak SSO provider
+[Dex](#dex-options) | [Obeject] | Configuration options for Dex SSO provider
```
Comment (reginapizza) [inline docs/reference/argocd.md:1236]: typo in object

File: docs/SECURITY.md [hunk 2]
```diff
@@ -18,4 +18,5 @@ Security vulnerabilities will be disclosed via release notes. Release notes for
 [GitHub Security Advisories](https://github.com/argoproj-labs/argocd-operator/security/advisories)
 feature to keep our community well informed, and will credit you for your findings (unless you prefer to stay anonymous, of course).
 
-**Note**: Please report the vulnerabilities related to Argo CD operator only. Please read Argo CD security [policy](https://github.com/argoproj/argo-cd/blob/master/SECURITY.md) if you would like to report any vulnerabilities related to Argo CD.
+!!! note
```
Comment (reginapizza) [inline docs/SECURITY.md:21]: Not an issue with your code necessarily, but this page does not have an associated link in the side nav, so there is no way for a user to find this page unless they search for it or manually input the url for it.

**Review by reginapizza — COMMENTED** [review]
Just a couple nits, other looks good :)

File: docs/reference/argocd.md (hunk 1 above)
Comment (jaideepr97) [reply docs/reference/argocd.md:1236]: fixed, thanks!

File: docs/SECURITY.md (hunk 2 above)
Comment (jaideepr97) [reply docs/SECURITY.md:21]: fixed, thanks!

**Comment by wtam2018** [comment]
LGTM

**Review by wtam2018 — APPROVED** [review]

**Pull request merged**